Improve Config Reference docs (#42)
DefinedNet/nebula-docs: docs/config/listen.mdx

sidebar_position: 4
---

import { Pill } from '@components/Pill/Pill';

# listen

> `listen` sets the UDP port Nebula will use for sending/receiving traffic and for handshakes. For a lighthouse node, the port should be defined, conventionally to `4242`, however using port 0 will dynamically assign a port and is recommended for roaming nodes.

Reviewer: I think we should pull the `listen.port` specific information down to the `listen.port` area, maybe we can simplify the first sentence as well?

> `listen` sets the UDP port Nebula will use for sending and receiving all traffic.

Author: Done

## listen.host

<Pill className="mb-24">Default: 0.0.0.0</Pill>

`host` is the ip of the interface to use when binding the listener. The default is 0.0.0.0, which is what most people should use.

## listen.port

> `port` is the UDP port nebula should use on a host. Setting this to `0` will dynamically assign a port number.

Reviewer: Then here we can say something like

> `port` is the UDP port nebula should use on a host. Setting this to `0` will dynamically assign a port number. Using `0` on lighthouses and relay hosts will likely lead to connectivity issues.

Author: Good point w/ the relays/lighthouses

## batch

> <Pill className="mb-24">Default: 64</Pill>

Reviewer: There are a handful of OS specific settings, maybe a future improvement is to make a pill for it?

Author: What would you put in such a thing? It's not entirely clear what all of the config values that exist are from the example config file in slackhq/nebula

> Sets the max number of packets to pull from the kernel for each syscall (under systems that support `recvmmsg`). Does not support reload

Reviewer 2: ```suggestion
Sets the max number of packets to pull from the kernel for each syscall (on systems that support `recvmmsg`).
```

We don't call out which settings can or cannot be reloaded currently, so I think it's a bit confusing to add the information to this setting.  We should probably find a good way to document all such settings, in a separate PR.
